client/src/boot/apollo/getGraphqlFragments.js

```diff
@@ -20,8 +20,23 @@ const handleError = response => {
   return response;
 };
 
-const getGraphqlFragments = async (baseUrl) => {
-  const typesUrl = `${baseUrl}admin.types.graphql`;
+const getGraphqlFragments = async (baseUrl, preferStatic = true) => {
+  // Backward compatibility hack. Remove when GraphQL 4 is in core
+  const isLegacy = !!document.body.getAttribute('data-graphql-legacy');
+
+  const urls = isLegacy
+    ? [
+      `${baseUrl}assets/types.graphql`,
+      `${baseUrl}graphql/types`,
+    ]
+    : [
+      `${baseUrl}admin.types.graphql`,
+    ];
+
+  if (!preferStatic) {
+    urls.reverse();
+  }
+  const [primaryURL, fallbackURL] = urls;
```

Comment:
Won't fallback URL be undefined for graphql4?  Seems a bit weird that we'll `response = await fetchSchema(fallbackURL);` below